crdschurch/crds-components
# DE7895: [Rebrand] Home Paged logged out cookie issue

This PR resolves an issue where a `nf_jwt` cookie token can be leftover when a user logs out. This stems from a change in how the `nf_jwt`  cookie is added. At some point, the cookie domain was changed for how it is being stored as `.{HOSTNAME}` example `.int.crossroads.net` or `{HOSTNAME}` example `int.crossroads.net`. I believe it was changed back leaving the other cookie this change will remove both no matter what.

This fix was implemented previously but had `.crossroads.net` hardcoded. This PR adds the ability to remove it in any environment.

[Rally Story](https://rally1.rallydev.com/#/66096747656d/custom/24109743995?qdp=%2Fdetail%2Fdefect%2F401025139156)

## How to test

**Important:** You will need to test locally on `local.crossroads.net`.
**Important:** Please verify it works on `int.crossroads.net` before green checking in code review.

1. Log in to either **int** or **prod** depending on what the value of `CRDS_ENV` is in your `.envrc`
1. Visit any component on the main landing page in another tab.
1. Add a second `nf_jwt` cookie with the domain `.local.crossroads.net` (the value does not matter)
    ![screenshot 2020-07-08 at 09 59 14](https://user-images.githubusercontent.com/56852436/86931171-c6a63500-c105-11ea-94b7-d47c46dfd39b.png)
1. Refresh the page to make sure the cookie is not removed while logged in **(If it is removed make sure you logged into the right environment)**
1. Sign out of either **int** of **prod**.
1. Refresh the components page
1. Verify that both cookies have been removed.
